signinapp/auth.py

```diff
@@ -146,6 +146,39 @@ def register_mentor():
     return render_template("auth/register.html.jinja2", form=form)
 
 
+@auth.route("/register/guardian", methods=["GET", "POST"])
+def register_guardian():
+    form = UserForm()
+    form.password.flags.is_required = True
+    del form.admin_data
+    del form.student_data
+    del form.subteam
+
+    if form.validate_on_submit():
+        # Cannot use form.populate_data because of the password
+        user = Guardian.get_from(
```

Comment:
Can we / Does this error if we don't find the user?

Reply:
Using `get_from` will create the user if it doesn't exist

Reply:
Ah right, I guess that's a perfectly valid workflow for this.